math/geometry.h

```diff
@@ -358,6 +387,212 @@ bool FurthestFreePointCircle(const Eigen::Matrix<T, 2, 1>& line_start,
+    return min_dist;
+  };
+
+  // l0 inside the circle, l1 outside the circle.
+  if (l0_dir.squaredNorm() < Sq(a_radius) &&
+      l1_dir.squaredNorm() >= Sq(a_radius)) {
+    const auto direction_line =
+        GetNormalizedOrZero(Eigen::Matrix<T, 2, 1>(l1 - l0));
+    if (direction_line_intersects_arc(proj_center_line,
+                                      along_line_dist,
+                                      a_angle_start,
+                                      a_angle_end,
+                                      a_center,
+                                      rotation_sign,
+                                      direction_line)) {
+      // Arc intersects with line.
+      return 0;
+    }
+
+    return no_arc_intersect_min_distance(
+        a_center, a_angle_start, a_angle_end, a_radius, l0, l1);
+  }
+
+  // l1 inside the circle, l0 outside the circle.
+  if (l0_dir.squaredNorm() >= Sq(a_radius) &&
+      l1_dir.squaredNorm() < Sq(a_radius)) {
+    const auto direction_line =
+        GetNormalizedOrZero(Eigen::Matrix<T, 2, 1>(l0 - l1));
+    if (direction_line_intersects_arc(proj_center_line,
+                                      along_line_dist,
+                                      a_angle_start,
+                                      a_angle_end,
+                                      a_center,
+                                      rotation_sign,
+                                      direction_line)) {
+      // Arc intersects with line.
+      return 0;
+    }
+    return no_arc_intersect_min_distance(
+        a_center, a_angle_start, a_angle_end, a_radius, l0, l1);
+  }
+
+  // Both l0 and l1 are not in the circle, but the line goes through the circle.
+  const auto direction_line =
+      GetNormalizedOrZero(Eigen::Matrix<T, 2, 1>(l1 - l0));
+
+  const bool positive_direction_intersect =
+      direction_line_intersects_arc(proj_center_line,
+                                    along_line_dist,
+                                    a_angle_start,
+                                    a_angle_end,
+                                    a_center,
+                                    rotation_sign,
+                                    direction_line);
+  const bool negative_direction_intersect =
+      direction_line_intersects_arc(proj_center_line,
+                                    along_line_dist,
+                                    a_angle_start,
+                                    a_angle_end,
+                                    a_center,
+                                    rotation_sign,
+                                    -direction_line);
+
+  if (positive_direction_intersect || negative_direction_intersect) {
+    // Arc intersects with line.
+    return 0;
+  }
+
+  const T min_distance = no_arc_intersect_min_distance(
+      a_center, a_angle_start, a_angle_end, a_radius, l0, l1);
+  return min_distance;
+}
+
+template <typename T>
+util::Pose FollowTrajectory(const util::Pose& pose_global_frame,
```

Comment:
Can you please explain what this function is supposed to do? If it's robot code, rather than general geometry code, it should go in a different file.

Reply:
That was an oversight on my part. I have removed it.